equinix/fabric_network_schema.go

```diff
@@ -0,0 +1,148 @@
+package equinix
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var createNetworkChangeRes = &schema.Resource{
+	Schema: createNetworkChangeSch(),
+}
+
+func createNetworkChangeSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"href": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "href",
+		},
+		"uuid": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "UUID of Network",
+		},
+		"type": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "network type",
+		},
+	}
+}
+
+var createNetworkOperationSchRes = &schema.Resource{
+	Schema: createNetworkOperationSch(),
+}
+
+func createNetworkOperationSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"equinix_status": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Network operation status",
+		},
+	}
+}
+
+func createNetworkProjectSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"project_id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Optional:    true,
+			Description: "Project Id",
+		},
+		"href": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Description: "Unique Resource URL",
+		},
+	}
+}
+
+func createNetworkResourceSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"href": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Fabric Network URI information",
+		},
+		"name": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Fabric Network name. An alpha-numeric 24 characters string which can include only hyphens and underscores",
+		},
+		"state": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Fabric Network overall state",
+		},
+		"scope": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Fabric Network scope",
+		},
+		"equinix_asn": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Equinix ASN",
+		},
+		"type": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringInSlice([]string{"IPWAN", "EPLAN", "EVPLAN"}, true),
+			Description:  "Supported Network types - EVPLAN, EPLAN, IPWAN",
+		},
+		"location": {
```

Comment:
If the value can be input into the schema or returned from the API response, then it needs to have Optional and Computed set to true.

Reference for what behaviors to include is explained in detail here: https://equinixjira.atlassian.net/wiki/spaces/IC/pages/145918564397/Terraform+Schema+Standards